src/error.rs

```diff
@@ -9,14 +11,19 @@ pub enum Error {
     Http(http::Error),
     HttpStatus(http::StatusCode),
     Json(serde_json::Error),
-    AuthError(AuthError),
+    Auth(AuthError),
     #[cfg(feature = "jwt")]
-    InvalidRsaKey,
+    InvalidRsaKey(Unspecified),
+    #[cfg(feature = "jwt")]
+    InvalidRsaKeyRejected(KeyRejected),
 }
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        use Error::*;
+        use Error::{
+            Auth, Base64Decode, Http, HttpStatus, InvalidKeyFormat, InvalidRsaKey,
+            InvalidRsaKeyRejected, Io, Json,
+        };
```

Comment:
The idea behind the lint that was triggered by this is good, but I don't think it makes sense when this is the same file the type is actually defined in, so I think I'd prefer if you just disabled that particular lint for this file.